Wait ability end task

Frickerson opened the pull request.

Commit:
Create task

Commit:
Fixup task

Add optional asc.
Remove destroy.

TheEmidee commented on Source/GASExtensions/Private/Tasks/GASExtAT_WaitAbilityEnd.cpp:
```diff
@@ -0,0 +1,32 @@
+
+void UGASExtAT_WaitAbilityEnd::OnAbilityEnded( const FAbilityEndedData & ability_ended_data )
+{
+    if ( ability_ended_data.AbilitySpecHandle == AbilitySpecHandle )
```
look at `UGASExtAT_WaitCharacterLanded`

check `if ( ShouldBroadcastAbilityTaskDelegates() )`

you can also add a flag `bTriggerOnce` amd call `EndTask` if true

Unbind from the event in `OnDestroy`

Frickerson replied:
Done.

TheEmidee commented on Source/GASExtensions/Public/Tasks/GASExtAT_WaitAbilityEnd.h:
```diff
@@ -0,0 +1,36 @@
+
+    void Activate() override;
+
+protected:
```
can be `private` (especially with a `final` class)

Frickerson replied:
Done.

Commit:
Merge remote-tracking branch 'origin/develop' into wait_ability_end_task

Commit:
PR fixes

Add bool for triggering once, defaulted to true.
check for ShouldBroadcastAbilutyTaskDelegates.
Add ondestroy to remove the binding.
Call Endtask if trigger once is true.

Commit:
Revert "Merge remote-tracking branch 'origin/develop' into wait_ability_end_task"

This reverts commit a09d3ae4ed89710e721e6c9ff83cc2418f367376, reversing
changes made to fe6a1867fc06e47294a0adf9450dead378f1ef70.
